lib/money/allocator.rb

```diff
@@ -132,15 +162,16 @@ def all_rational?(splits)
       splits.all? { |split| split.is_a?(Rational) }
     end
 
-    def allocation_index_for(strategy, length, idx)
-      case strategy
-      when :roundrobin
-        idx % length
-      when :roundrobin_reverse
-        length - (idx % length) - 1
-      else
```

Comment:
as a note - we didn't re-implement the `% length` in the `def allocate` above as it should be mathematically impossible to have more `left_over`s to re-distribute than we have `amounts.length`.

The `left_over` value should sum up to the amounts `fractionnal_subunits`